lib/system.nim

```diff
@@ -52,6 +52,15 @@ type
   typed* {.magic: Stmt.}         ## Meta type to denote an expression that
                                  ## is resolved (for templates).
 
+# XXX: the ``EnumToStr`` magic has to be defined here so that the ``bool``
+#      definition can compile
```

Comment:
This feels better, because the cost of the complexity of the `system` module's current arrangement is now paid for by the `system` module.